src/api/updateFiles.ts

```diff
@@ -401,3 +417,139 @@ const checkResultsForErrors = (results: DocumentBulkResponse[]): void => {
+
+  // Map each document key to its StoreDocument
+  const storeDocumentMap = documentResults.reduce<Map<string, StoreDocument>>(
+    (storeDocumentMap, result) => {
+      // Handle error cases
+      if ('error' in result) {
+        if (result.error !== 'not_found') {
+          throw result
+        }
+        return storeDocumentMap
+      }
+
+      // Set the document key to map to the StoreDocument
+      storeDocumentMap.set(result.key, result.doc)
+
+      // Return map accumulator
+      return storeDocumentMap
+    },
+    new Map()
+  )
+
+  // Iterate over each document key and check if its timestamp is consistent
+  // with its pointer timestamp in the parent document.
+  for (const [parentKey, childKeys] of documentKeyGraph.entries()) {
+    const doc = storeDocumentMap.get(parentKey)
+    const parentStoreDirectoryDocument = asMaybe(asStoreDirectoryDocument)(doc)
+
+    for (const childKey of childKeys) {
+      const [, childDocumentPath] = childKey.split(':')
+      const childDocmentName = getNameFromPath(childDocumentPath)
+
+      // Get the file pointer timestamp for the child document
+      const filePointerTimestamp =
+        parentStoreDirectoryDocument != null
+          ? parentStoreDirectoryDocument.paths[childDocmentName]
+          : undefined
+      // Get the deleted file pointer timestamp for the child document
+      const deletedFilePointerTimestamp =
+        parentStoreDirectoryDocument != null
+          ? parentStoreDirectoryDocument.deleted[childDocmentName]
+          : undefined
+      // Get the child document
+      const childDoc = storeDocumentMap.get(childKey)
+
+      if (filePointerTimestamp == null) {
+        // This is a new document if both file pointer and document aren't present
+        if (childDoc == null) continue
+        // This is a deleted document if deleted file pointer is present
+        if (deletedFilePointerTimestamp != null) continue
+        // Otherise, inconsistent if no file pointer for child document
+        return false
+      }
+
+      // Inconsistent if child doc is missing while a file pointer is present
+      if (childDoc == null) return false
+
+      const childStoreFileDocument = asMaybe(asStoreFileDocument)(childDoc)
+      const childStoreDirectoryDocument = asMaybe(asStoreDirectoryDocument)(
+        childDoc
+      )
+
+      if (childStoreFileDocument != null) {
+        // Inconsistent if pointer timestamp and document timestamp don't match
+        if (!eq(filePointerTimestamp, childStoreFileDocument.timestamp))
+          return false
+        continue
+      }
+      if (childStoreDirectoryDocument != null) {
+        // Inconsistent if pointer timestamp and document timestamp don't match
+        if (!eq(filePointerTimestamp, childStoreDirectoryDocument.timestamp))
+          return false
+        continue
+      }
+
+      throw new Error(`Unexpected document type during repo consistency check`)
+    }
+  }
+
+  return true
```

Comment:
Either throw or return an object with some exposition into why it is inconsistent.